CommentAgent.py

```diff
@@ -314,7 +316,9 @@ def add_pr_messages(self) -> None:
         """
         Adds messages to the GPTWrapper instance related to the pull request.
         """
```

Comment:
The filtering of valid file extensions should be separate from pulling data from the API. Consider filtering the file extensions within the `get_pr_deltas` method itself to keep the `add_pr_messages` method more generic and easier to test.
        ```suggestion
        pr_deltas = self.GH.get_pr_deltas(self.pr)
        pr_deltas = [delta for delta in pr_deltas if delta.endswith(tuple(GPTsettings.FILE_EXTENSIONS))]
        ```